src/jmc/compile/command/utils.py

```diff
@@ -596,13 +598,24 @@ def __parse_bracket(self) -> None:
                     raise JMCValueError(
                         f"Unknown property '{prop_}'", self.token, self.tokenizer)
                 key, json_body, is_local = self.datapack.data.formatted_text_prop[prop_]
-                if not callable(json_body):
+                if isinstance(json_body, dict) and 'nbt' in json_body.keys():
+                    self.current_json[key] = {}
+                    self.current_json[key]["nbt"] = json_body["nbt"](arg) # type: ignore
+                    self.current_json[key]["interpret"] = json_body["interpret"](arg) # type: ignore
+                    if "entity" in json_body.keys():
+                        self.current_json[key]["entity"] = json_body["entity"] # type: ignore
+                    if "block" in json_body.keys():
+                        self.current_json[key]["block"] = json_body["block"] # type: ignore
+                    if "storage" in json_body.keys():
+                        self.current_json[key]["storage"] = json_body["storage"] # type: ignore
+                elif not callable(json_body):
                     raise JMCValueError(
                         f"Custom property '{prop_}' expected no argument", self.token, self.tokenizer, suggestion="Remove '()'")
-                if not arg:
+                elif not arg:
                     raise JMCValueError(
                         f"Expected value inside parenthesis of property '{prop_}' (got nothing)", self.token, self.tokenizer)
-                self.current_json[key] = json_body(arg)
+                else:
+                    self.current_json[key] = json_body(arg)
```

Comment:
it is nested into a condition
not sure if it will cause any errors

Reply:
seems reasonable